### DISCUSSION TITLE
(Ubuntu 18.04, SoftConsole v2021.1) Programming error - fpgenprog

### DISCUSSION THREAD

**Discussion by okawaj9** [polarfire-soc/polarfire-soc-documentation#95] [HSS and U-Boot] [answered]
Confirmation - Below is expected solution against HSS programming error ?

- OS: Ubuntu 18.04
- SoftConsole: v2021.1
- HSS revision: master as of June 20th
- Programming operation: non-secure boot mode 1
<img width="419" alt="aaa" src="https://user-images.githubusercontent.com/61904969/122674300-77222180-d20f-11eb-8436-21ff53bed4c4.png">

- HSS programming error message on the SoftConsole:
======================================
21:23:26 ERROR - Failed to detect fpgenprog in various locations. Please set the FPGENPROG environment variable to fully qualified fpgenprog executable path. fpgenprog should be located in your Program Debug Tool or Libero SoC installation.
======================================

**[Solution]** : Adding below into softconsole.sh in SoftCosole install directory 
export FPGENPROG="[Libero SoC v2021.1 Install Directory]/Libero/bin64/fpgenprog"

**Comment by hughbreslin** [comment, marked as answer]
Correct 🙂
